ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/MessageScrubber.java

```diff
@@ -46,10 +50,10 @@ public static boolean shouldScrub(Map<String, String> attributes, JSONObject jso
             || ("beta".equals(attributes.get(Attribute.APP_UPDATE_CHANNEL))
                 && attributes.get(Attribute.APP_VERSION).startsWith("68")))
         && Optional.of(json) // payload.metadata.RemoteType
-            .map(j -> j.optJSONObject("payload")) //
-            .map(j -> j.optJSONObject("metadata")) //
-            .map(j -> j.optString("RemoteType")) //
-            .filter(s -> s.startsWith("webIsolated=")).isPresent()) {
+            .map(j -> j.get("payload")) //
+            .map(j -> j.get("metadata")) //
+            .map(j -> j.get("RemoteType")) //
+            .map(JsonNode::textValue).filter(s -> s.startsWith("webIsolated=")).isPresent()) {
```

Comment:
`JsonNode::get` throws a runtime error if the key doesn't exist, and `JsonNode::textValue` will return `null` if the node isn't a string.
I think we should use `JsonNode::path` and `Optional::ofNullable` instead:
```java
Optional.ofNullable(json.path("payload").path("metadata").path("RemoteType").textValue())
    .filter(s -> s.startsWith("webIsolated=")).isPresent()
```

Reply:
I was under the impression that `JsonNode::get` would return null in the case of a missing element according to docs for that method.

But I did not know about `path` and indeed that looks easier for chained calls.

I was relying on `JsonNode::textValue` to return null in the case it's not a string, which will return `Optional.empty` for a `map` call on an `Optional` instance.